File: leetcode-rs/1200_1299/leetcode-no1221.rs
// problem: https://leetcode.cn/problems/split-a-string-in-balanced-strings/description/
struct Solution;

impl Solution {
    // 0 1 1 
    pub fn tribonacci(n: i32) -> i32 {
        let mut base = vec![0, 1, 1, 2];
        for i in 0..n - 2 {
            base[3] = base[2] + base[1] + base[0];
            base[0] = base[1];
            base[1] = base[2];
            base[2] = base[3];
        }
        match n {
            0 => { 0 }
            1 => { 1 }
            _ => { base[2] }
        }
    }
}

fn main() {
    println!("{:?}", Solution::tribonacci(0));
    println!("{:?}", Solution::tribonacci(1));
    println!("{:?}", Solution::tribonacci(2));
    println!("{:?}", Solution::tribonacci(3));
    println!("{:?}", Solution::tribonacci(4));
    println!("{:?}", Solution::tribonacci(5));
    println!("{:?}", Solution::tribonacci(25));
}

//
// Created by zhiya at 2025/5/24 22:24
//

File: leetcode-cs/1200_1299/leetcode-no1221.cs
// problem: https://leetcode.cn/problems/split-a-string-in-balanced-strings/?envType=problem-list-v2&envId=3rBpGyas 

namespace leetcodeNo1221;

using System;

public class Solution {
    public int BalancedStringSplit(string s) {
        int state = 0;
        int res = 0;
        for (int i = 0; i < s.Length; ++i) {
            state += (s[i] != 'R') ? 1 : -1;
            res += state == 0 ? 1 : 0;
        }

        return res;
    }
}

public class Test {
    public static void Run() {
        var s = new Solution();
        Console.WriteLine(s.BalancedStringSplit("RLRRLLRLRL"));
        Console.WriteLine(s.BalancedStringSplit("RLRRRLLRLL"));
        Console.WriteLine(s.BalancedStringSplit("LLLLRRRR"));
    }
}

//
// Created By jing At 11/3/25 22:11:56
//